SCons/Executor.py

```diff
@@ -306,30 +306,30 @@ def get_all_children(self):
         over and over), so removing the duplicates once up front should
         save the Taskmaster a lot of work.
         """
-        result = SCons.Util.UniqueList([])
+        result = []
         for target in self.get_all_targets():
             result.extend(target.children())
-        return result
+        return SCons.Util.uniquer_hashables(result)
```

Comment:
You'll get a (miniscule) extra gain by importing `uniquer_hashables` at the top and avoiding the `SCons.Util` lookup in these three methods.

Reply:
I thought about doing something similar. But my expectation was that most consumers are going to iterate the full return from these functions, and after some discussion with @bdbaddog he hinted that the generator based approach (which I believe `chain` is) are mostly useful when you aren't actually going to pull all the results and are mostly after it for the laziness.

Reply:
How about:
`
all_targets = self.get_all_targets()
return SCons.Util.uniquer_hashables([item for sublist in all_targets for item in sublist])
`

Comprehensions are often significantly faster than for loops..
see this: (the accepted answer) https://stackoverflow.com/questions/952914/how-do-i-make-a-flat-list-out-of-a-list-of-lists

Reply:
I tried it, as:

```
-        result = []
-        for target in self.get_all_targets():
-            result.extend(target.children())
-        return SCons.Util.uniquer_hashables(result)
+        return SCons.Util.uniquer_hashables([child for target in self.get_all_targets() for child in target.children()])
```

But I didn't see any further improvement. Which I don't think is surprising. The gains here are from 1) not allowing the `UniqueList` to become the object iterated, and 2) slightly improving the de-duplication. I don't think the production of the list to be de-duplicated was ever part of the cost here.